taichi/transforms/scalarize.cpp

```diff
@@ -281,15 +281,55 @@ class Scalarize : public BasicStmtVisitor {
         Stmt *print_stmt = std::get<Stmt *>(content);
         if (print_stmt->is<MatrixInitStmt>()) {
           auto matrix_init_stmt = print_stmt->cast<MatrixInitStmt>();
-          for (size_t j = 0; j < matrix_init_stmt->values.size(); j++) {
-            new_contents.push_back(matrix_init_stmt->values[j]);
+          auto tensor_shape =
+              print_stmt->ret_type->as<TensorType>()->get_shape();
+
+          bool is_matrix = tensor_shape.size() == 2;
+          int m = tensor_shape[0];
+          int n = is_matrix ? tensor_shape[1] : 1;
+
+          new_contents.push_back("[");
+          if (is_matrix) {
+            for (size_t i = 0; i < m; i++) {
+              new_contents.push_back("[");
+              for (size_t j = 0; j < n; j++) {
+                size_t index = i * n + j;
+                new_contents.push_back(matrix_init_stmt->values[index]);
+                new_contents.push_back(", ");
```

Comment:
```suggestion
    if (j != n - 1)
                new_contents.push_back(", ");
```
I believe here we should check if we are processing the last element?